railties/lib/rails/test_unit/runner.rb

```diff
@@ -61,7 +61,9 @@ def compose_filter(runnable, filter)
         private
           def extract_filters(argv)
             # Extract absolute and relative paths but skip -n /.*/ regexp filters.
-            argv.select { |arg| %r%^/?\w+/%.match?(arg) && !arg.end_with?("/") }.map do |path|
+            argv.reject { |arg| regexp_filter?(arg) }
+                .select { |arg| path_argument?(arg) }
```

Comment:
Would it be clearer as `select { |arg| path_argument?(arg) && !regexp_filter?(arg) }`?

Reply:
I tried it out, and yes, it would 😄 fits on one line too!